vis.go

```diff
@@ -4,10 +4,13 @@ package mtree
 import "C"
 import "fmt"
 
+// Vis is a wrapper of the C implementation of the function vis, which encodes
+// a character with a particular format/style
 func Vis(str string) (string, error) {
-	dst := new(C.char)
+	// dst needs to be 4 times the length of str (plus 1 for null char)
+	dst := C.CString(string(make([]byte, 4*len(str)+1)))
 	ret := C.strvis(dst, C.CString(str), C.VIS_WHITE|C.VIS_OCTAL|C.VIS_GLOB)
-	if ret == 0 {
```

Comment:
now I haven't confirmed this, but I think this return number is the length of the dest string. I'm wondering if there is actually an error condition ..